src/user/user.service.ts

```diff
@@ -103,6 +103,27 @@ export class UserService {
     return elderly;
   }
 
+  async checkRelationship(eid: number, cid: number) {
+    let elderly;
+    let isRelated = false;
+    try {
+      elderly = await this.findElderlyById(eid);
+    } catch {
+      throw new NotFoundException('Elderly Does Not Exist');
+    }
+      
+    const caretakers = await this.findCaretakerByElderlyId(elderly.uid);
+    if (!caretakers) return isRelated;
+
+    for (var i = 0; i < caretakers.length; i++) {
+      if (caretakers[i].uid === cid) {
+        isRelated = true
+        break;
+      }
+    }
```

Comment:
can be replaced with 
```suggestion
 const isRelated = !!caretakers.find((caretaker) => caretaker.uid===cid);
```